### PR TITLE
[issue#1510] fix - NN_TORCH model is incorrectly handling crashing during inference because no batchnorm statistics recorded

### PR DESCRIPTION
**Pull request by gradientsky**
*Issue #, if available:*
https://github.com/awslabs/autogluon/issues/1510 -  NN_TORCH incorrectly implements use_batchnorm=True

*Description of changes:*
Removed `track_running_stats=False` from batchnorm layers, which is forcing always using batch statistics. This fails with a single row. Default will preserve running statistics from the training time.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by Innixma** [comment]
@jwmueller feel free to leave any comments. Does this look correct?

**Comment by szha** [comment]
Job PR-1514-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1514/1/index.html

**Review by jwmueller — APPROVED** [review]
Good catch, tested and it appears to work properly now!  
Not sure why those lines got added into the batchnorm layers in the first place...

**Comment by gomlfx** [comment]
Does GluonTS have an email digest thread?

**Review by Innixma — APPROVED** [review]
LGTM!

**Comment by gomlfx** [comment]
That would be nice if it weren't that most discussions are unanswered even
after months.
But don't worry I heard you the first time. I won't write here again if not
directly related to the PR or Github issue. Have a nice day.
[image: image.png]

**Comment by Innixma** [comment]
> That would be nice if it weren't that most discussions are unanswered even after months.

The grand majority of GitHub issues are responded to and addressed. Of the 605 total issues, I have personally [authored](https://github.com/awslabs/autogluon/issues?q=is%3Aissue+author%3Ainnixma) or [commented](https://github.com/awslabs/autogluon/issues?q=is%3Aissue+commenter%3Ainnixma) on 501 of them. The majority of discussions are also answered or replied to outside of vision related questions, as we currently lack bandwidth to dedicate towards vision questions and are planning a major refactor to address many such questions / functionality requests at once rather than trying to fix them one at a time within the constraints of a shaky foundation. I sincerely do appreciate your engagement, and am simply suggesting that for the future benefit of users seeking answers to the same questions you are asking to make a dedicated issue/discussion so that it is easier to find.

**Comment by gomlfx** [comment]
Ok then.
